projects/smoke_bts7040/src/main.c

```diff
@@ -0,0 +1,111 @@
+// A simple smoke test for BTS7040
+
+// Periodically take reading from user selected channels and log the result
+// Configurable items: wait time, FRONT or REAR power distro selection, channels to be tested
+
+#include "bts7040_load_switch.h"
+#include "controller_board_pins.h"
+#include "current_measurement_config.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "mux.h"
+#include "pca9539r_gpio_expander.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// experimentally determined value: scaling factor of 670/1160 ~ 0.578 ohms, bias of -8 mA
+#define SMOKE_BTS7040_SENSE_RESISTOR 1160
+#define SMOKE_BTS7040_BIAS (-8)
+
+#define SMOKE_BTS7040_MIN_FAULT_VOLTAGE_MV 3200
+#define SMOKE_BTS7040_MAX_FAULT_VOLTAGE_MV 10000
+
+// Smoke test settings. Can be modified to fit testing purpose.
+#define CURRENT_MEASURE_INTERVAL_MS 500  // Set wait time between each set of readings
+#define IS_FRONT_POWER_DISTRO true       // Set whether to test FRONT or REAR power distro
+
+// Maximum number of channels that can be tested,
+// which is the same number of bts7200 used in front/rear power distribution
+#define MAX_TEST_CHANNELS 8
```

Comment:
There's only 6 BTS7040s though, eh? Since there's only 6 entries in each `bts7040s` array in the hw configs, if you try to index past 5 with a channel of 6 or 7, you'll get a segfault or other undefined behaviour. So this should be 6. (Also change the comment from bts7200 to bts7040.)